### ISSUE TITLE
Ask about the operating platform

### ISSUE THREAD

**Issue by HarrisHongting** [andvg3/LSDM#4]
Hi An,

May I know the platform you used while working on this project and the corresponding version? I'm having different problems when trying to run this project with both WSL2 and Ubuntu 24.04 in VirtualBox. 

Many thanks,

Harris

**Comment by andvg3** [comment]
Hi @HarrisHongting ,

For training models, I think it is possible (though I have never tested) to run on your specified system. However, it is recommend to run _directly_ on Ubuntu 22.04 (or above). Could you please try to run on a proper Linux system? Additionally, for training, it is possible to run on headless Linux system; however, for visualization, a Linux system with attached display _must_ be utilized.

Best regards, An Vuong.

**Comment by HarrisHongting** [comment]
Hi @andvg3 ,

Thank you for getting back to me. I finally installed the Linux system (Ubuntu 24.04) on my computer and tried to rerun this project without these problems.

When I tried to run:
python -m run.test_sdm data/protext/proxd_test/ \
--load_model training/sdm/model_ckpt/best_model_cfd.pt \
--model_name sdm \
--fix_ori \
--test_on_valid_set \
--output_dir training/sdm/output

Terminal reminds:

PolyMeshT::add_face: patch re-linking failed
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
envs/lsdm/lib/python3.8/site-packages/scipy/sparse/_index.py:103: SparseEfficiencyWarning: Changing the sparsity structure of a csr_matrix is expensive. lil_matrix is more efficient.
self._set_intXint(row, col, x.flat[0])
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge
PolyMeshT::add_face: complex edge

It looks like it encountered a problem while trying to build the polygon mesh. All model files are downloaded from the PRO-teXt link provided in the readme text. But the code works. Should I just ignore these errors or is there any way to fix this?

Many thanks,

Harris
